### ISSUE TITLE
Allow automatic generation of function signature in function XML comments

### ISSUE THREAD

**Issue by cmdoffing** [fsharp/fslang-suggestions#1384]
**I propose we allow automatic generation of function signature in function XML comments. 

```fsharp
/// Some function description comment.
/// <signature>
let f x y = ...
```

** The signature is the most important piece of documentation and it should be included. The existing way of approaching this problem in F# is to manually write out the signature. Doing this automatically ensures that we save time and get it right.

## Pros and Cons 

**The advantages of making this adjustment to F# are** better documentation and saved time.

**The disadvantages of making this adjustment to F# are** time to implement and an extra line in the function docs.

## Extra information

**Estimated cost (XS, S, M, L, XL, XXL):**  
S. Automatic signature generation is already done in FSI, so it should be an easy implementation.

**Related suggestions:** (put links to related suggestions here)

## Affidavit (please submit!)

Please tick these items by placing a cross in the box:
* [x] This is not a question (e.g. like one you might ask on [StackOverflow](http://stackoverflow.com)) and I have searched StackOverflow for discussions of this issue
* [x] This is a language change and not purely a tooling change (e.g. compiler bug, editor support, warning/error messages, new warning, non-breaking optimisation) belonging to [the compiler and tooling repository](https://github.com/dotnet/fsharp)
* [x] This is not something which has obviously "already been decided" in previous versions of F#. If you're questioning a fundamental design decision that has obviously already been taken (e.g. "Make F# untyped") then please don't submit it
* [x] I have [searched both open and closed suggestions on this site](http://github.com/fsharp/fslang-suggestions/issues) and believe this is not a duplicate

Please tick all that apply:
* [x] This is not a breaking change to the F# language design
* [ ] I or my company would be willing to help implement and/or test this

## For Readers

If you would like to see this issue implemented, please click the :+1: emoji on this issue. These counts are used to generally order the suggestions by engagement.

**Comment by Tarmil** [comment]
Tooltips should be perfectly capable of displaying the signature without having to put it in the doc comment, manually or automatically. What editor are you using?

**Comment by vzarytovskii** [comment]
> Tooltips should be perfectly capable of displaying the signature without having to put it in the doc comment, manually or automatically. What editor are you using?

Yeah, I second that. XML docs is usually complimentary to the signature in tooltips in tooling or used in the autogenerated documentation (like we have for fslib). What would be some examples when you need an additional signature in the xmldoc itself?

**Comment by cmdoffing** [comment]
My bad! I haven't used tooltips with my own XML docs. I didn't realize
there was a separate mechanism being used to generate the signature in
Visual Studio 2022, which I use. Thanks a lot for your rapid response and
please close the request.

-- CMD

**Comment by voronoipotato** [comment]
This does vaguely remind me of the perennial haskell style inline sigs generation suggestion. Basically the same as the FSI signatures file but placed prior to each function. The value is slightly low but it could be nice when you're trying to look at code in github.

**Comment by cmdoffing** [comment]
Thanks for the tip!

**Comment by T-Gro** [comment]
(closing, this is supported by tooling already)
